src/utils/commands.ts

```diff
@@ -1,13 +1,6 @@
 import { commands } from "../../types/common";
 
-import {
-  destDir,
-  user,
-  winUser,
-  windestDir,
-  tempDIR,
-  baseURL,
-} from "./globals";
+import { destDir, user, tempDIR, baseURL, platform, root } from "./globals";
 
 export const codespacesCommands = (
```

Comment:
What is the purpose of leaving the codespacesCommands in here 🤔 I see you have generalised the windows/mac/Linux commands, what is the purpose of not doing the codespaces either 🤔 is it because we call this function specifically if it's codespaces. 

I see quite a few of the command objects are reused across the `codespacesCommands` and `generalCommands`. If we are going down your approach (which I like by the way) it may be worth exporting each command as it's own object and then referencing the commands, it would save us copying the same command syntax for codespaces and general, if that makes sense? 

E.G

```
export const gitCloneCommand = { ... }

---

import { gitCloneCommand } from "./"

```

Does that make sense? I love the idea of going general, but I think we can go a step further and do the same for codespaces. Maybe create a directory called `commands` and then have a single file in each directory for each command. Import all of them commands into the `index.ts` in the `commands` directory, then just import the commands you need into the `src/utils/commands.ts` file 👍 

Let me know if that makes sense 👍

Reply:
Absolutely agree, codespace could be consolidated. 

It's been a minute since I've looked at it and struggling to remember why I didn't include it. I thought I put it in the commit log, but I don't see it. 

At a glance, I think it had to do with the cwd being different. There's no sense of a user path, which made it unique. Now that I think about it root could just include user if it's linux, MacOS or windows. Also, I'm not sure how it works but do you get an entirely new instance each time you use it? Do you even need to rm any files/folders?

In the end I think it was a time constraint and that I've never worked with codespace.

Not sure I follow on `Maybe create a directory called commands and then have a single file in each directory for each command`. Do you mean each and every command or based on git vs OS commands?

I'm a bit of a map lover and would maybe mix the two. Something like below. I'm more proficient in groovy so my example will be in that and without writing out the entire thing...

```
branch = 'ghas-test'
platform = 'darwin'
osPlatform = (platform == 'win32') ? platform : 'nix'
cwd = 'Users/djthornton1212/Desktop/test'

osCommands = [
    mkdir : [
         win32: 'mkdir',
         nix: 'mkdir -p'
     ],
     rm : [
        win32: 'rmdir /Q /S',
        nix: 'rm -rf'
     ],
     copy: [
        win32: 'copy',
        nix: 'cp'
     ]
]

gitCommands = [
    clone: [
         general: 'git clone',
         darwin: 'git clone --depth 1 --filter=blob:none --sparse'
    ],
    checkout: [
        general: "git checkout -b ${branch}"
    ],
    add: [
        general: 'git add'
    ],
    commit: [
        general: 'commit -m "Commit CodeQL File"'
     ],
    push: [
        general: "git push --set-upstream origin ${branch}"
     ]
]

ghasCommands = [
     (osCommands.rm."$osPlatform" + " $cwd"),
     (osCommands.mkdir."$osPlatform" + " $cwd"),
     (platform == 'darwin' ? gitCommands.clone.darwin : gitCommands.clone.general),
     gitCommands.checkout.general,
     gitCommands.add.general,
     gitCommands.commit.general,
     gitCommands.push.general,
     (osCommands.rm."$osPlatform" + " $cwd"),
]

for (command in ghasCommands ) {
    println "executing: $command"
}
```

This way we just export commandOrder... I don't know need to think that over some.

Reply:
Okay, let's focus on this:

> Absolutely agree, codespace could be consolidated.

don't worry about the second part, I think it's fine as it is 🙇 But I think it would be good to consolidate down the commands to Codespaces as well 👍 

> Also, I'm not sure how it works but do you get an entirely new instance each time you use it? Do you even need to rm any files/folders?

Great question, each codespace is a fresh instance, meaning for everyone you create, you get new folders, etc. However, the codespace can last for however long you need to. You can reuse the same codespace, but most people just start from scratch. 

It would be super easy to test, just [create a new org](https://github.com/account/organizations/new?plan=free&ref_cta=Create%2520a%2520free%2520organization&ref_loc=cards&ref_page=%2Forganizations%2Fplan) (it's free) and then create three repos:

1. The repository where you will run the script from (this is where you will create the codespace from)
2. A Javascript repo with a single file in called `hello.js` 
3. Another Javascript repo with a single file in called `hello.js` 

Then just follow the steps [here](https://github.com/NickLiffen/ghas-enablement#running-this-within-a-codespace) and run the script. It should be super simple, Codespaces are nice and quick to get up and running. 

That would allow you to test your changes quick and easily :) 

If you could do this, then I think it would make this change 💯

Reply:
@NickLiffen Quick update, though organizations are free, codespaces aren't. 

> Codespaces is available for organizations using GitHub Team or GitHub Enterprise Cloud. For more information,

I'm discussing with my management about using it in our organization, but at the moment don't have it available to me.  Can you test that component once I update the PR?

Reply:
Yes, of course, I think you would like Codespaces if you can use it, but if you need me to test, I am happy to for sure 🙇